### PR TITLE
Bug fix - redirect route rule to egress and additional tests to pilot's integration test

### PR DESCRIPTION
**Pull request by ronenschafferibm**
The bug is that when configuring a redirect route rule to egress there is a redundant 'cluster' key in envoy's routes config.

Additional tests to the e2e test:
- Redirect route rule to egress
- Rewrite route rule to egress
- Fault injection to egress traffic configured with wildcard

### CODE REVIEW

File: pilot/proxy/envoy/config.go [hunk 1]
```diff
@@ -887,19 +896,10 @@ func buildEgressVirtualHost(rule *routing.EgressRule,
...
 	// reset the protocol to the original value
 	port.Protocol = protocolToHandle
 
-	if len(routes) > 0 {
-		// Set the destination clusters to the cluster we computed above.
-		// Services defined via egress rules do not have labels and hence no weighted clusters
-		for _, route := range routes {
-			route.Cluster = externalTrafficCluster.Name
-			route.clusters = []*Cluster{externalTrafficCluster}
-		}
```
Comment (rshriram) [inline pilot/proxy/envoy/config.go]: I think you should keep this, and simply `nil` out any weighted cluster in the route. This way, all the custom egress processing is confined inside this function.

File: pilot/proxy/envoy/discovery_test.go [hunk 2]
```diff
@@ -496,6 +496,16 @@ func TestRouteDiscoveryRedirect(t *testing.T) {
 	}
 }
 
+func TestRouteToEgressDiscoveryRedirect(t *testing.T) {
```
Comment (rshriram) [inline pilot/proxy/envoy/discovery_test.go]: TestRouteDiscoveryEgressRedirect.. Or better still, fold this test case into the function above (TestRouteDiscoveryRedirect), where you add another egressRule and egressRedirectConfig

File: pilot/proxy/envoy/route.go [hunk 3]
```diff
@@ -228,9 +228,8 @@ func buildHTTPRouteV1(config model.Config, service *model.Service, port *model.P
 		}
 	} else {
 		// default route for the destination
-		cluster := buildOutboundCluster(destination, port, nil)
-		route.Cluster = cluster.Name
-		route.clusters = append(route.clusters, cluster)
+		route.Cluster = defaultCluster.Name
+		route.clusters = append(route.clusters, defaultCluster)
```
Comment (rshriram) [inline pilot/proxy/envoy/route.go]: Thsi does not look right.. The abstraction seems broken if we compute clusters for weighted clusters but not for the non weighted case.

You can overwrite the computed cluster at the call site as i described above, unless there is a major restriction

File: pilot/test/integration/routingToEgress.go [hunk 4]
```diff
@@ -41,22 +44,51 @@ func (t *routingToEgress) setup() error {
...
+			configRouting: "rule-fault-injection-to-egress.yaml.tmpl",
+			routingData: map[string]string{
+				"service": "httpbin.org",
+			},
 			check: func() error {
 				return t.verifyFaultInjectionByResponseCode("a", "http://httpbin.org", 418)
 			},
 		},
+		{
+			description:   "inject a http fault in traffic to *.httpbin.org",
```
Comment (rshriram) [inline pilot/test/integration/routingToEgress.go]: This is nice!

File: pilot/test/integration/routingToEgress.go [hunk 5]
```diff
@@ -65,15 +97,117 @@ func (t *routingToEgress) run() error {
...
+				"from":      "/post",
+				"to":        "/get",
+				"authority": "httpbin.org",
+			},
+			check: func() error {
+				return t.verifyEgressRedirectRewrite("a", "http://httpbin.org/post", "httpbin.org", "/get")
+			},
+		},
+		{
+			description:   "redirect traffic from httpbin.org/post to *.httpbin.org/get",
```
Comment (rshriram) [inline pilot/test/integration/routingToEgress.go]: Nicely done.. Thanks for these tests!

**Comment by vadimeisenbergibm** [comment]
Submitted by #2621.

**Pull request closed without merging**